src/main/java/io/cdap/plugin/gcp/spanner/sink/SpannerSink.java

```diff
@@ -133,14 +134,19 @@ public void prepareRun(BatchSinkContext context) throws Exception {
   }
 
   private void createTableIfNotPresent(DatabaseClient dbClient, Database database,
-                                       Schema schema) throws ExecutionException, InterruptedException {
+                                       @Nullable Schema schema) throws ExecutionException, InterruptedException {
     boolean tableExists = isTablePresent(dbClient);
     if (!tableExists) {
       if (Strings.isNullOrEmpty(config.getKeys())) {
         throw new IllegalArgumentException(String.format("Spanner table %s does not exist. To create it from the " +
                                                            "pipeline, primary keys must be provided",
                                                          config.getTable()));
       }
+      if (schema == null) {
```

Comment:
this check can be moved to line 103

Reply:
This check can't be here since it only needs to be checked if table not exist, if table already exist, we do not need to enforce the schema field